query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/converters/PinotFunctionConverter.java

```diff
@@ -62,6 +76,21 @@ private String functionToString(
     return function.getFunctionName() + "(" + argumentString + ")";
   }
 
+  private String functionToStringForAvgRate(
+      Function function,
+      java.util.function.Function<Expression, String> argumentConverter,
+      ExecutionContext executionContext) {
+
+    List<Expression> argumentsList = new ArrayList<>();
```

Comment:
So I responded to the other comment before re-reviewing this - this is pretty close, the only change I'd really suggest is again the removal of the arg list and working with the values more directly (converting each extracted arg to a string then interpolating the final result, rather than converting to different args and building a joined single arg string and interpolating that).